Assets/Tests/InputSystem/CoreTests_Actions.cs

```diff
@@ -1834,6 +2119,658 @@ public void Actions_CanReadValueFromAction_AsButton(InputActionType actionType,
+            // |--------|-------------|------------------|-----|--------------|--------|
+            // | Value  | Default     | W -> S, P, S*    | T/  |              |        |
+            // | Value  | Press       | W -> S           |  /  | true         |        |
+            // | Value  | Hold        | W    (No Change) |  /  |              | true   |
+            // | Button | Default     | W -> S           |  /  | true         |        |
+            // | Button | Press       | W -> S           |  /  | true         |        |
+            // | Button | Hold        | W    (No Change) |  /  |              | true   |
+            // | Pass   | Default     | W -> P           | T/  |              |        |
+
+            Assert.That(action.IsPressed(), Is.False);
+            Assert.That(action.WasPressedThisFrame(), Is.False);
+            Assert.That(action.WasReleasedThisFrame(), Is.False);
+
+            if (action.type == InputActionType.Value && interactions == null)
+            {
+                Assert.That(trace, Started(action).AndThen(Performed(action)));
+                Assert.That(action.phase, Is.EqualTo(InputActionPhase.Started));
+                Assert.That(action.WasPerformedThisFrame(), Is.True);
+                Assert.That(action.WasUnperformedThisFrame(), Is.False);
+            }
+            else if (isButtonLike)
+            {
+                Assert.That(trace, Started(action));
+                Assert.That(action.phase, Is.EqualTo(InputActionPhase.Started));
+                Assert.That(action.WasPerformedThisFrame(), Is.False);
+                Assert.That(action.WasUnperformedThisFrame(), Is.False);
+            }
+            else if (isHold)
+            {
+                Assert.That(trace, Is.Empty);
+                Assert.That(action.phase, Is.EqualTo(InputActionPhase.Waiting));
+                Assert.That(action.WasPerformedThisFrame(), Is.False);
+                Assert.That(action.WasUnperformedThisFrame(), Is.False);
+            }
+            else if (action.type == InputActionType.PassThrough)
+            {
+                Assert.That(trace, Performed(action));
+                Assert.That(action.phase, Is.EqualTo(InputActionPhase.Performed));
+                Assert.That(action.WasPerformedThisFrame(), Is.True);
+                Assert.That(action.WasUnperformedThisFrame(), Is.False);
+            }
+
+            trace.Clear();
+
+            // Press some more such that it crosses the press threshold.
+            Set(gamepad.leftTrigger, 0.75f);
+
+            // W = Waiting, S = Started, P = Performed, C = Canceled
+            // P/U = Performed/Unperformed
+            // (* means listeners not invoked)
+            // | Type   | Interaction | Phase Change     | P/U | isButtonLike | isHold |
+            // |--------|-------------|------------------|-----|--------------|--------|
+            // | Value  | Default     | S -> P, S*       | T/  |              |        |
```

Comment:
What is the meaning of "T/" could this be explained in incline comment? Does it mean Performed=true, Unperformed=false? In that case maybe T/F is easier to understand?

Reply:
Yes it's True/False. I kept it blank because I thought it added visual noise and made it harder to see which ones are true at a glance, but I've switched it to F everywhere it was blank before. I also added an explanation to the legend above.